### PR TITLE
upgrading setuptools and pip in patch-image

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
To fix ci.
Ref: https://jenkins.nordix.org/view/Metal3%20Periodic/job/metal3_daily_main_fullstack_building/191/consoleFull

### CODE REVIEW

File: patch-image.sh [hunk 1]
```diff
@@ -18,9 +18,12 @@ while IFS= read -r line; do
     git fetch "$PROJ_URL" "$REFSPEC"
     git checkout FETCH_HEAD
 
+    /usr/bin/python3 -m pip install --upgrade pip
+    pip3 install -U setuptools
```
Comment (elfosardo) [inline patch-image.sh:22]: same here
if we need setuptools, let's include it in dnf command in L7

File: patch-image.sh [hunk 2]
```diff
@@ -18,9 +18,12 @@ while IFS= read -r line; do
...
+
     SKIP_GENERATE_AUTHORS=1 SKIP_WRITE_GIT_CHANGELOG=1 python3 setup.py sdist
     pip3 install --prefix /usr dist/*.tar.gz
 done < "$PATCH_FILE"
 
-dnf remove -y python3-pip git-core
+dnf remove -y python3-pip git-core setuptools
```
Comment (elfosardo) [inline patch-image.sh:28]: youre removing setuptools with dns when it was installed with pip?

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: patch-image.sh [hunk 3]
```diff
@@ -18,9 +18,12 @@ while IFS= read -r line; do
     git fetch "$PROJ_URL" "$REFSPEC"
     git checkout FETCH_HEAD
 
+    /usr/bin/python3 -m pip install --upgrade pip
```
[the remark this answers is not in the document]
Comment (adilGhaffarDev) [reply patch-image.sh:21]: We don't need to upgrade pip, I will resolve the problem by upgrading the setuptools version. It seems that the error that we are getting in our in patch-image script is caused by a change in the setuptools version. Previously, when rpm was the default package manager, we were using python3-setuptools-57.4.0. However, in recent CI runs, we are now getting python3-setuptools-53.0.0. I'm unsure why this version is changing. I am running local tests where I solely upgrade the setuptools for the patch image script, I will push those changes once test passed. Holding this PR

[the remark this answers is not in the document]
Comment (elfosardo) [reply patch-image.sh:21]: that's odd, the same setuptools is used during installation in https://github.com/metal3-io/ironic-image/blob/main/prepare-image.sh#L31
it's then removed, but you can reinstall it using dnf

**Comment by adilGhaffarDev** [comment]
fixed the issue by removing patching. https://github.com/Nordix/metal3-dev-tools/pull/681 
I will add followup pr to remove patch script too.

**Pull request closed without merging**